### PR TITLE
re-construct unit test structure and add namespace for oss module

### PR DESCRIPTION
**Pull request by Rembrant777**
in this commit
re-organize oss unit test class structure add test class to encapsulate all test cases in module granularity

### CODE REVIEW

File: oss/ossSocket.hpp [hunk 1]
```diff
@@ -25,136 +25,136 @@
 #define OSS_MAX_HOSTNAME NI_MAXHOST 
 #define OSS_MAX_SERVICENAME NI_MAXSERV 
 
-class _ossSocket {
```
Comment (Rembrant777) [inline oss/ossSocket.hpp:28]: to better maintain and isolate modules we add namespace emeralddb/${module_name}.
all modules should be under the namespace of emeralddb

File: test/oss_test.cpp [hunk 2]
```diff
@@ -20,120 +20,172 @@
...
+using namespace testing;
+
+
+class ossSocketTest : public Test {
+public:
+  ossSocketTest() {}
+  ~ossSocketTest() {}
+
+protected: 
+  virtual void SetUp() {
```
Comment (Rembrant777) [inline test/oss_test.cpp:34]: here the SetUp/TearDown two virtual methods that are both decorated by protected are two methods inherited from gtest's class Test. we can override it by adding more context information however cannot modify their methods' signature

File: test/oss_test.cpp [hunk 3]
```diff
@@ -20,120 +20,172 @@
...
+    _port = genRandomPort(); 
+    _timeout = 6; 
+    ossPtr = getInstance(); 
+  }
+
+  virtual void TearDown() {
+    cout << "after test" << endl; 
+    release(ossPtr);
+    ossPtr = nullptr; 
+    ASSERT_EQ(nullptr, ossPtr);
```
Comment (Rembrant777) [inline test/oss_test.cpp:45]: gtest supports `ASSERT_EQ` `ASSERT_TRUE` `ASSERT_FALSE` `EXPECT_EQ` not support `ASSERT_NQ` 
also `ASSERT_LT` `ASSERT_GT`

File: test/oss_test.cpp [hunk 4]
```diff
@@ -20,120 +20,172 @@
...
+    cout << "setup server hostname [" << hostnamePtr << "] is listening to [" << _port << "]" << endl; 
+  }
+
+    void release(ossSocket *ossPtr) {
+    if (ossPtr != nullptr) {
+      free(ossPtr);
+    }
+  }
+
+  void connect() {
```
Comment (Rembrant777) [inline test/oss_test.cpp:86]: in this test case, only validate that the client side and server side's connection can be established successfully, but still require communication like exchanging message test case scenario to verify whether the communication is ok.

what's more, we also need to design some more corner cases like exchanging more data packages exchanging 
that the size of the message package is too large for server/client side to handle. 

or set a high threshold connection to the server side and then create multiple connection requests from the client side to the server side in which the server-side handle too many connection requests to handle, until it gets an exception and exits.
